readers-frontend/src/pages/[year]/[month]/[day]/[slug].astro

```diff
@@ -85,31 +51,11 @@ export async function getStaticPaths() {
         article.categories
       ).categories_id;
 
-      // Get 4 most recent articles in parent category
-      const mSimilarArticlesIds = (
-        await prisma.articles.findMany({
-          where: {
-            articles_categories: {
-              contains: parentCategoryId.toString(),
-            },
-            articles_id: {
-              not: article.articles_id, // Exclude same article
-            },
-          },
-          take: 4,
-          select: {
-            articles_id: true,
-          },
-          orderBy: {
-            articles_published: "desc",
-          },
-        })
-      ).map((similarArticleId) => similarArticleId.articles_id); // Only take IDs
-
-      const mSimilarArticles = await getArticles(mSimilarArticlesIds);
-
-      accumulator.set(article.articles_id, mSimilarArticles);
+      if (!accumulator.has(parentCategoryId)) {
+        const mSimilarArticles = await getSimilarArticles(parentCategoryId);
 
+        accumulator.set(parentCategoryId, mSimilarArticles);
```

Comment:
> Just wondering can't we remove this accumulator's logic to be inside of the map below. So in the map you would get the parent category id and the similar articles?

This allows to map them by category so the query happens only once per category. If we moved it into the array map you'd still need a `Map<number, articlesWithArticleDrafts[]>` to only query for category rather than each article